app/models/application_record.rb

```diff
@@ -2,6 +2,7 @@ class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
 
   include Purgeable
+  extend UniqueAcrossModels
```

Comment:
nit/optional: as we spoke, this wasn't what I was suggesting, but I'll leave it up to you whether to keep or undo this refactor.

Reply:
@maestromac is the approach you had suggested described anywhere? It would be interesting to see.

Reply:
This is shorter, but we only need `UniqueAcrossModels` for 4 models, why extend all of them?